R/dl-cur-hosp-data.R

```diff
@@ -41,69 +41,60 @@ current_hosp_data <- function() {
 
     # URL for file
     url <- "https://data.cms.gov/provider-data/sites/default/files/archive/Hospitals/current/hospitals_current_data.zip"
-    f_name <- basename(url)
 
-    # Create a temporary file to store the zip file
-    tmp <- tempfile()
+    # Create a temporary directory to process the zip file
     tmp_dir <- tempdir()
+    download_location <- file.path(tmp_dir, "download.zip")
+    extract_location <- file.path(tmp_dir, "extract")
 
     # Download the zip file to the temporary location
     utils::download.file(
         url = url,
-        destfile = tmp
+        destfile = download_location
     )
 
     # Unzip the file
-    utils::unzip(tmp, exdir = tmp_dir)
+    utils::unzip(tmp, exdir = extract_location)
 
     # Read the csv files into a list
     csv_file_list <- list.files(
-        path = tmp_dir,
+        path = extract_location,
         pattern = "\\.csv$",
         full.names = TRUE
     )
 
+    # make named list
+    csv_names <-
+        stats::setNames(
+            object = csv_file_list,
+            nm =
+                csv_file_list |>
+                basename() |>
+                gsub(pattern = "\\.csv$", replacement = "") |>
+                janitor::make_clean_names()
+        )
+
```

Comment:
If you pass a named vector to `lapply()` or `map()` it will name them as you go. I added the step to remove the ".csv" extension in the table names and used `janitor` since it's already in your namespace.

Reply:
I like  this, I will need to update other files to account for this as they look for explicit names but ultimately with these I think dropping the file extension is nicer.